Scripts/RemediationScripts/Remediate-EnableSecurityScanningForContainerRegistry.ps1

```diff
@@ -0,0 +1,716 @@
+        $validcontainerRegistryResources = $containerRegistryResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+        
+        $validcontainerRegistryResources| ForEach-Object {
+            $resourceId = $_.ResourceId
+
+            try
+            {
+                $containerRegistryResource =  Get-AzResource -ResourceType Microsoft.ContainerRegistry/registries -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+                $containerRegistryDetails += $containerRegistryResource
+            }
+            catch
+            {
+                Write-Host "Error fetching Container Registry(s) resource: Resource ID - $($resourceId). Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            }
+        }
+    }
+
+    $totalContainerRegistry = ($containerRegistryDetails| Measure-Object).Count
+
+    if ($totalContainerRegistry -eq 0)
+    {
+        Write-Host "No Container Registry(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+  
+    Write-Host "Found [$($totalContainerRegistry)] Container Registry(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+                                                                          
+    Write-Host $([Constants]::SingleDashLine)
+    # List for storing role assignment details.
+    $roleAssignmentDetails = @()
+
+    # List for storing sub level scanning identity role assignment details.
+    $subLevelRoleAssignmentDetails = @()
+
+    Write-Host "Fetching central reader security scanner role assignments..."
+
+    # Fetching all role assignment.
+    $roleAssignmentDetails = Get-AzRoleAssignment 
+
+    # Seperating the central account. 
+    $centralReaderAccounts = $roleAssignmentDetails | Where-Object {$_.RoleDefinitionName -eq "Reader" -and $_.ObjectId -eq $ObjectId} 
+
+    Write-Host "Found [$(($centralReaderAccounts| Measure-Object).Count)] central reader security scanner role assignment(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # list for storing Container Registry(s) for which central reader role is not assigned.
+    $containerRegistryWithoutSecurityScanningEnabled = @()
+
+    Write-Host "Separating Container Registry(s) for which access is not granted to security scanning identities..."
+
+    #seperating the sub level role assignments.
+    $subLevelRoleAssignmentDetails = $centralReaderAccounts | Where-Object {$_.Scope -eq "/subscriptions/$($SubscriptionId)"}
+
+    if(($subLevelRoleAssignmentDetails| Measure-Object).Count  -eq 0)
+    {
+        $containerRegistryDetails | ForEach-Object {
+            $containerRegistry = $_
+            $CentralReaderAccount = $centralReaderAccounts | Where-Object {$_.Scope -eq $containerRegistry.ResourceId -or(($_.Scope).Split('/').Count -eq 5 -and ($_.Scope).Split('/')[4] -eq $containerRegistry.ResourceGroupName)}
```

Comment:
When checking for RG Scope also check if scope has resourceGroup bcz MG scope will also have same 5 count.